### PR TITLE
Create Tutorial 21 for PromptNode

### LINKED ISSUE #112: New tutorial for: `PromptNode` Basics
**Describe the tutorial you would like to see here**
I will create a second issue for a more advanced tutorial. However, for the `PromptNode` basics I would suggest we start with:
- Initializing a PromptNode with various models
- The default `PromptTemplates`
- Defining your own `PromptTemplate`

@agnieszka-m please add any input about how this can be 'task' oriented. 

**Additional context**
I think using the `PromptNode` in a pipeline is slightly more advanced and could be too distracting to have an intro tutorial that covers this. So I suggest a separate one for that

[x] I've checked the existing tutorials

### PR DESCRIPTION
**Pull request by bilgeyucel**
fixes #112 

[Preview Link](https://haystack-home-jf6hbezn0-deepset-overnice.vercel.app/tutorials/21_customizing_promptnode) on Haystack Website

### CODE REVIEW

**Comment by agnieszka-m** [comment]
hey @bilgeyucel , just had a peek in here. So I have a major problem with this - it's not really a tutorial. It's a guide, and if you look at what we currently have for prompt node in our docs, it's exactly that - different usage examples. For a tutorial, we want people to build something concrete. So my suggestion here would be: let's add the examples you came up with here (and whatever else is missing) in the PromptNode guide https://docs.haystack.deepset.ai/docs/prompt_node and for a tutorial, let's think about a concrete use case. So for a proper tutorial that follows the structure we agreed on, we'd need:
- a concrete dataset people can download (or eventually replace with their own dataset)
- a pipeline that uses prompt node to do something with the dataset
I think it's crucial that we come up with a use case that's realistic and shows the potential of the prompt node. We could even have two tutorials - one for a beginner, where the usecase is simpler and they use one of the out of the box templates, and one for more advanced users where they create their own promptemplate, or something.

So yes, my two cents. Happy to talk more about it.

**Comment by bilgeyucel** [comment]
> hey @bilgeyucel , just had a peek in here. So I have a major problem with this - it's not really a tutorial. It's a guide, and if you look at what we currently have for prompt node in our docs, it's exactly that - different usage examples. For a tutorial, we want people to build something concrete. So my suggestion here would be: let's add the examples you came up with here (and whatever else is missing) in the PromptNode guide https://docs.haystack.deepset.ai/docs/prompt_node and for a tutorial, let's think about a concrete use case. So for a proper tutorial that follows the structure we agreed on, we'd need:
> 
> * a concrete dataset people can download (or eventually replace with their own dataset)
> * a pipeline that uses prompt node to do something with the dataset
>   I think it's crucial that we come up with a use case that's realistic and shows the potential of the prompt node. We could even have two tutorials - one for a beginner, where the usecase is simpler and they use one of the out of the box templates, and one for more advanced users where they create their own promptemplate, or something.
> 
> So yes, my two cents. Happy to talk more about it.

Hey @agnieszka-m, thank you for your comment. For the first PromptNode tutorial, I was thinking of not including the pipeline concept as it makes understanding the PromptNode a bit harder. However, I agree with you. Now, it looks more like a guide, and with a pipeline, it's easier to come up with a real-life use case. 
I'll try to come up with a very basic use case of PromptNode in the pipeline and make this tutorial an intermediate one rather than a beginner. I'll let you know about the progress 👍

**Comment by TuanaCelik** [comment]
@agnieszka-m and @bilgeyucel 
The `PromptNode` is a very flexible node with a lot of moving parts. And to be able to use it in a pipeline there's a lot of concepts to understand, first and foremost: the `Shaper` - to be able to use it properly.
I strongly recommend splitting the `PromptNode` tutorial into levels that achieve different things.
For the first tutorial, I suggest focusing on using the PromptNode on its own, and defining your own templates. This will allow the learner to see how to define your own `PromptNode` with your own template. There's already a good amount to cover there: the template you define can have arguments, that become arguments to the nodes `run()` function..

If we go for the pipeline in the first tutorial there's way too much to cover in one tutorial: Templates, custom templates, shaper, creating the pipeline.

That's why I suggested having this beginner tutorial cover:
The node
Pre defined templates
Creating custom templates

We can decide on what the best way to name this tutorial would be. Some suggestions:
- Create a PromtTemaplate
- PromptNode Basics: Define your own PromptTemplate

It may be similar to the guide in the docs, but people cannot run the code blocks in the docs and the Office Hours discussions have proved many times that we would benefit by having a tutorial like this to guide people through a simple prompt node setup.

On the note of datasets: I suggest we pick a simple set of examples, document, that can be used as an input to a `PromptTemplate`

A second, more advanced tutorial could cover a full pipeline, where we would have to show the `Shaper` as well..

File: markdowns/21_Customizing_PromptNode.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
+---
+layout: tutorial
+featured: True
+colab: https://colab.research.google.com/github/deepset-ai/haystack-tutorials/blob/main/tutorials/21_Customizing_PromptNode.ipynb
+toc: True
+title: "Customizing PromptNode for NLP Tasks"
+last_updated: 2023-02-15
+level: "intermediate"
+weight: 57
+description: Use PromptNode and PromptTemplate for your custom NLP tasks
```
Comment (TuanaCelik) [inline markdowns/21_Customizing_PromptNode.md:10]: I think this is a great title. Specific, and to the point 👍

**Review by TuanaCelik — APPROVED** [review]

**Comment by agnieszka-m** [comment]
Thank you @bilgeyucel for preparing this! I've added my 2cents (or even 5PLN ;). I've changed the language and titles here and there to make it more focused on what you can achieve with the node and the templates, rather than on describing the node itself. If you have any doubts about my changes, let me know, I'll be happy to explain my reasons.
Also, I took the liberty to change the imports to more general ones (e.g. instead of from haystack.nodes.prompt to from haystack.nodes) as I was recently told more general imports are generally recommended. So if you could please double-check this bit, that would be swell ;)

**Pull request merged**